File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteJobProducts/QuoteJobProducts.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace QuoteJobProductss

Public Class QuoteJobProducts

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "QuoteJobProducts Properties"

      
Private _QuoteJobProductsID As Integer = 0
Public Readonly Property QuoteJobProductsID() As Integer
	Get 
		Return _QuoteJobProductsID
	End Get	
End Property	
Public Writeonly Property SetQuoteJobProductsID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteJobProductsID", Value)
	End Set
End Property


Private _QuoteJobID As Integer = 0
Public Readonly Property QuoteJobID() As Integer
	Get 
		Return _QuoteJobID
	End Get	
End Property	
Public Writeonly Property SetQuoteJobID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteJobID", Value)
	End Set
End Property


Private _ProductID As Integer = 0
Public Readonly Property ProductID() As Integer
	Get 
		Return _ProductID
	End Get	
End Property	
Public Writeonly Property SetProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ProductID", Value)
	End Set
End Property


Private _Quantity As Integer = 0
Public Readonly Property Quantity() As Integer
	Get 
		Return _Quantity
	End Get	
End Property	
Public Writeonly Property SetQuantity() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Quantity", Value)
	End Set
End Property


Private _SellPrice As Double = 0
Public Readonly Property SellPrice() As Double
	Get 
		Return _SellPrice
	End Get	
End Property	
Public Writeonly Property SetSellPrice() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_SellPrice", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteJobProducts/QuoteJobProducts.cs
using System.Collections;

namespace FSM.Entity
{
    namespace QuoteJobProductss
    {
        public class QuoteJobProducts
        {
            private DataTypeValidator _dataTypeValidator;

            public QuoteJobProducts()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _QuoteJobProductsID = 0;

            public int QuoteJobProductsID
            {
                get
                {
                    return _QuoteJobProductsID;
                }
            }

            public object SetQuoteJobProductsID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteJobProductsID", value);
                }
            }

            private int _QuoteJobID = 0;

            public int QuoteJobID
            {
                get
                {
                    return _QuoteJobID;
                }
            }

            public object SetQuoteJobID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteJobID", value);
                }
            }

            private int _ProductID = 0;

            public int ProductID
            {
                get
                {
                    return _ProductID;
                }
            }

            public object SetProductID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ProductID", value);
                }
            }

            private int _Quantity = 0;

            public int Quantity
            {
                get
                {
                    return _Quantity;
                }
            }

            public object SetQuantity
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Quantity", value);
                }
            }

            private double _SellPrice = 0;

            public double SellPrice
            {
                get
                {
                    return _SellPrice;
                }
            }

            public object SetSellPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SellPrice", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}